actions.go

```diff
@@ -3,10 +3,157 @@ package neo4j
 import (
 	"fmt"
 
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
 	"github.com/cloudprivacylabs/opencypher/graph"
+	"github.com/fatih/structs"
 	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
 )
 
+type neo4jAction interface {
+	Run(neo4j.Transaction) error
+}
+
+var neo4jActions = map[string]neo4jAction{
+	"Address":     recreateEntity{},
+	"Observation": insert{},
+}
+
+type deleteEntity struct {
+	Config
+	graph.Graph
+	entityId int64
+}
+
+type createEntity struct {
+	Config
+	graph.Graph
+	graph.Node
+}
+
+type recreateEntity struct {
+	ce createEntity
+	de deleteEntity
+}
+
+func (r recreateEntity) Run(tx neo4j.Transaction) error {
+	err := r.de.delete(tx)
+	if err != nil {
+		return err
+	}
+	err = r.ce.create(tx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d deleteEntity) delete(tx neo4j.Transaction) error {
+	err := loadEntityNodes(tx, d.Graph, []int64{d.entityId}, d.Config, findNeighbors, nil)
+	if err != nil {
+		return err
+	}
+	// delete entity and all nodes reachable from this entity,
+	_, err = tx.Run("MATCH (m)-[*]->(n) WHERE ID(m)=$id DETACH DELETE m,n", map[string]interface{}{"id": d.entityId})
```

Comment:
This is going to delete everything reachable from m. You called loadEntityNodes, that should give you all the nodes and edges you should delete.